src/components/Common/UI/Badge/BadgeTypes.ts

```diff
@@ -0,0 +1,9 @@
+import type { HTMLAttributes } from 'react'
+
+export interface BadgeBaseProps
+  extends Pick<HTMLAttributes<HTMLSpanElement>, 'className' | 'id' | 'aria-label'> {
+  text: string
```

Comment:
I'm wondering if this should just be called `children` for consistency with other components. Generally we just use children as the content. We could also consider just loosening the type to `ReactNode`